### PR TITLE
WAITP-1439: mobile prompt

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Add util for detecting user browser and device type (mobile versus desktop)
- Update download app prompt to direct to the app if present, else the relevant app store
- Update download app prompt to store a 24 hour cookie so the user doesn't get spammed on every refresh

### CODE REVIEW

File: packages/datatrak-web/src/features/MobileAppPrompt.tsx [hunk 1]
```diff
@@ -75,54 +67,95 @@ const BROWSER_ICONS = {
...
+// each of these will have a fallback url and an app url. The app url will be a link to the app itself, and the fallback url will be a link to the applicable app store
 const APP_URL = {
-  ANDROID: 'https://play.google.com/store/apps/details?id=com.tupaiameditrak',
-  IOS: 'https://itunes.apple.com/us/app/tupaia-meditrak/id1245053537',
+  ANDROID: {
+    fallback: 'https://play.google.com/store/apps/details?id=com.tupaiameditrak',
+    app: 'intent://#Intent;package=com.tupaiameditrak;end', // eventually this is where the deep link will go
+  },
+  IOS: {
+    fallback: 'https://itunes.apple.com/us/app/tupaia-meditrak/id1245053537',
```
Comment (jaskfla) [inline packages/datatrak-web/src/features/MobileAppPrompt.tsx:77]: Perhaps safer to let the user’s device decide which App Store region to use?

```suggestion
    fallback: 'https://itunes.apple.com/app/tupaia-meditrak/id1245053537',
```

Comment (acdunham) [reply packages/datatrak-web/src/features/MobileAppPrompt.tsx]: Good point!

File: packages/datatrak-web/src/utils/detectDevice.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+// detect if the device is a mobile device (without relying on screen size)
+export const getIsMobileDevice = () => {
+  const regex = /Mobi|Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i;
```
Comment (tcaiger) [inline packages/datatrak-web/src/utils/detectDevice.ts:8]: Out of interest, where did you get the regex list? When I googled it the lists I saw were often much longer? eg. https://stackoverflow.com/questions/11381673/detecting-a-mobile-browser. Anyway overall I don't think it matters too much as if we don't prompt one user once in a while.

**Review by tcaiger — APPROVED** [review]
Looks great - very nicely organised and implemented

File: packages/datatrak-web/src/utils/detectDevice.ts (hunk 2 above)
Comment (acdunham) [reply packages/datatrak-web/src/utils/detectDevice.ts:8]: Also from a stackoverflow. I think it should theoretically pickup most mobile devices based on this one though

Comment (jaskfla) [reply packages/datatrak-web/src/utils/detectDevice.ts:8]: If from SO, remember to attribute! SO answers are shared under the [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0) licence, and it’s super easy to forget that really we should be crediting + linking, and indicating if we made changes

**Pull request merged**